src/components/Articles/Articles.js

```diff
@@ -1,52 +1,37 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import moment from 'moment';
+
+// import get all articles action
+import { getAllArticle } from '../../actions/article'
+
 // components
 import Article from '../Article/Article';
+
 // styles
 import styles from './articles.module.scss'
+
 // images
 import thumbnailImage from '../../demo.png';
 
 class Articles extends Component {
   constructor(props) {
     super(props);
-    this.state = {
-      articles: [
-        {
-          id: 1,
-          title: 'This Article talks about some Tall Buildings And City Night Life',
-          slug: 'this-is-slug',
-          description: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
-          thumbnail: thumbnailImage,
-        },
-        {
-          id: 2,
-          title: 'This Article talks about some Tall Buildings And City Night Life',
-          slug: 'this-is-slug',
-          description: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
-          thumbnail: thumbnailImage,
-        },
-        {
-          id: 3,
-          title: 'This Article talks about some Tall Buildings And City Night Life',
-          slug: 'this-is-slug',
-          description: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
-          thumbnail: thumbnailImage,
-        },
-        {
-          id: 4,
-          title: 'This Article talks about some Tall Buildings And City Night Life',
-          slug: 'this-is-slug',
-          description: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
-          thumbnail: thumbnailImage,
-        }
-      ]
-    }
+    this.state = {};
+  }
+
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch(getAllArticle());
   }
 
   render() {
-    const { articles } = this.state;
+    const { articles } = this.props;
+    const { data } = articles;
+    const data2 = data.slice(0);
```

Comment:
I think you can skip line `30` and just have the array of articles directly in your `articles` prop by parsing it in your `mapStateToProps` method as:
```
const mapStateToProps = state => {
  const { allArticleReducer } = state;
  return {
    articles: allArticleReducer.data,
  }
}
```

This will give you access to the array itself which you can then sort